### ISSUE TITLE
Is FLRE thread safe??

### ISSUE THREAD

**Issue by mWaltari** [BeRo1985/flre#80]
IF I do Following:

...
GFLRE := TFLRE.Create(C_REG_EXP_STRING):
...

Thread 1:
 GFLRE.MartchAll(StringDataOFThread1Contect)
 
 Thread 2:
  GFLRE.MartchAll(StringDataOFThread2Contect)
  
  So multiple thread would share TFLRE instance but would run it on data in threads own context.

**Comment by BeRo1985** [comment]
Yes, it should. This is also the reason, why whole the TFLREThreadLocalStorageInstance stuff exists.

**Comment by mWaltari** [comment]
If you could, put this on the readme.md, valuable information.

Thanks man!

-Tee-
